lib/cpp/test/ToStringTest.cpp

```diff
@@ -46,23 +46,31 @@ BOOST_AUTO_TEST_CASE(base_types_to_string) {
 // Windows:
 #ifndef _WIN32
 BOOST_AUTO_TEST_CASE(locale_en_US_int_to_string) {
+  try {
 #ifdef _WIN32
-  std::locale::global(std::locale("en-US.UTF-8"));
+    std::locale::global(std::locale("en-US.UTF-8"));
 #else
-  std::locale::global(std::locale("en_US.UTF-8"));
+    std::locale::global(std::locale("en_US.UTF-8"));
 #endif
-  BOOST_CHECK_EQUAL(to_string(1000000), "1000000");
+    BOOST_CHECK_EQUAL(to_string(1000000), "1000000");
+  } catch (const std::runtime_error& e) {
+
```

Comment:
Empty catch-all clauses are triggering some sort of "Are you sure?" reflex. Especially if the patch is called a fix. I would rather label it as "curing a symptom". Can we have at least a comment explaining why this is the optimal way to handle it?